Homework2

ceib opened the pull request:
Signed-off-by: Aleksey Vorontsov <avorontsov1987@gmail.com>

ceib committed:
Homework2

Signed-off-by: vorontsovas <vorontsovas@rainvest.ru>

IVaSmi commented on src/main/scala/collections/task_collections.scala:
```diff
@@ -15,8 +15,13 @@ object task_collections {
-    List.empty
+  def capitalizeIgnoringASCII(text: List[String]): List[String] = text match {
+    case ::(head, tail) =>
+      head :: tail.collect{
```
Хороший вариант решения:) 

Можно ещё так решить: 

```
 text.zipWithIndex.collect{
    case (str, 0) => str
    case (str, index) if isASCIIString(str) => str.toUpperCase
    case (str, index)  => str.toLowerCase
  }
```

IVaSmi commented on src/main/scala/collections/task_collections.scala:
```diff
@@ -15,8 +15,13 @@ object task_collections {
-  def capitalizeIgnoringASCII(text: List[String]): List[String] = {
-    List.empty
+  def capitalizeIgnoringASCII(text: List[String]): List[String] = text match {
+    case ::(head, tail) =>
```
Решение хорошее но сломается если на вход придёт List("o") например. Поправьте чтобы такого не было:)

IVaSmi replied:
`case head::tail 
`
Ещё так можно писать^

IVaSmi commented on src/main/scala/collections/task_collections.scala:
```diff
@@ -29,7 +34,24 @@ object task_collections {
+      "9" -> "nine",
+      "0" -> "zero"
+    )
+    val TenNumber: String = "10"
```
с 10 я ошибся в тесте, не планировалась она

IVaSmi replied:
Но можно её в Map поместить тоже)

IVaSmi commented on src/main/scala/collections/task_collections.scala:
```diff
@@ -29,7 +34,24 @@ object task_collections {
+    )
+    val TenNumber: String = "10"
+    val TenString: String = "ten"
+    text.replace(TenNumber, TenString).flatMap {
```
если решать только от 0 до 9, то решение через Char верное, а если решать через 0-бесконечность, то конечно уже придётся использовать String и делать строку по " "

ceib committed:
- В методе numbersToNumericString перенес 10 в мапу и сделал замену по словам с разделением по " ";
- Добавил метод numbersToNumericStringForSingleDigitsOnly, реализующий задачу только для числе 0-9 (перес сюда старый вариант numbersToNumericString, не предусматривающий возможность появления числе больше 10)
